source/agora/test/Base.d

```diff
@@ -2201,16 +2233,6 @@ public class NoPreImageVN : TestValidatorNode
             super.onPreImageRevealTimer();
     }
 
-    public override PreImageInfo getPreimage (Hash enroll_key)
-    {
-        if (enroll_key != this.enroll_man.getEnrollmentKey())
-            return super.getPreimage(enroll_key);
-        else if (atomicLoad(*this.reveal_preimage))
-            return super.getPreimage(enroll_key);
-        else
-            throw new Exception("No such PreImage");
-    }
-
```

Comment:
Yeah, that should not be removed. It's the whole point of the class.

Reply:
> Yeah, that should not be removed. It's the whole point of the class.
👀 
 
getPreimage (Hash enroll_key) 
was removed from the FullNode's interface...

However it is true that I should have added the new method called
getPreimageWhatever (Set!Hash enroll_keys)